clouddrift/sphere.py

```diff
@@ -435,3 +435,133 @@ def sphere_to_plane(
     y[..., :] = np.cumsum(dy, axis=-1)
 
     return x, y
+
+
+def earthsphere_to_cartesian(
+    lon: np.ndarray,
+    lat: np.ndarray,
+    earth_radius: Optional[float] = EARTH_RADIUS_METERS,
+) -> Tuple[np.ndarray, np.ndarray, np.ndarray]:
+    """Converts latitude and longitude on the spherical Earth to
+     Cartesian coordinates in three dimensions.
+
+    The Cartesian coordinate system is a right-handed system whose
+    origin lies at the center of the spherical Earth.  It is oriented with the
+    Z-axis passing through the poles and the X-axis passing through
+    the point lon = 0, lat = 0. This function is inverted by `cartesian_to_sphericalearth`.
+
+    Parameters
+    ----------
+    lon : np.ndarray
+        An N-d array of longitudes in degrees.
+    lat : np.ndarray
+        An N-d array of latitudes in degrees.
+    earth_radius: float, optional
+        The radius of the Earth in meters. Default is EARTH_RADIUS_METERS = 6.3781e6.
+
+    Returns
+    -------
+    x : np.ndarray
+        x-coordinates in 3D in meters.
+    y : np.ndarray
+        y-coordinates in 3D in meters.
+    z : np.ndarray
+        z-coordinates in 3D in meters.
+
+    Examples
+    --------
+    >>> earthsphere_to_cartesian(np.array([0, 45]),np.array([0, 45]))
+    (array([6378100., 3189050.]),
+    array([      0., 3189050.]),
+    array([      0.        , 4509997.76108592]))
+
+    >>> earthsphere_to_cartesian(np.array([0, 45, 90]),np.array([0, 90, 180]),earth_radius=1)
+    (array([ 1.00000000e+00,  4.32978028e-17, -6.12323400e-17]),
+    array([ 0.00000000e+00,  4.32978028e-17, -1.00000000e+00]),
+    array([0.0000000e+00, 1.0000000e+00, 1.2246468e-16]))
+
+    >>> x, y, z = earthsphere_to_cartesian(np.array([0,5]),np.array([0,5]))
+
+    Raises
+    ------
+    AttributeError
+        If ``lon`` and ``lat`` are not NumPy arrays.
+
+    See Also
+    --------
+    :func:`cartesian_to_earthsphere`
+    """
+
+    x = earth_radius * np.cos(lat * np.pi / 180) * np.cos(lon * np.pi / 180)
+    y = earth_radius * np.cos(lat * np.pi / 180) * np.sin(lon * np.pi / 180)
+    z = earth_radius * np.sin(lat * np.pi / 180)
```

Comment:
I think using numpy `np.radians` makes things cleaner.

```suggestion
    x = earth_radius * np.cos(np.radians(lat)) * np.cos(np.radians(lat))
    y = earth_radius * np.cos(np.radians(lat)) * np.sin(np.radians(lon))
    z = earth_radius * np.sin(np.radians(lat))
```